ssl/ssl_lib.cc

```diff
@@ -3047,6 +3047,45 @@ uint16_t SSL_get_peer_signature_algorithm(const SSL *ssl) {
   return session->peer_signature_algorithm;
 }
 
+int SSL_get_peer_signature_nid(const SSL *ssl, int *psig_nid) {
+  if(psig_nid == NULL) {
+    return 0;
+  }
```

Comment:
This is still a style preference but for future reference we have:
```suggestion
  GUARD_PTR(psig_nid);
```